src/regbot/fetch/drugsfda.py

```diff
@@ -68,12 +69,63 @@
 )
 
 
+def _map_to_enum(cls: type[Enum], value: str, mapping: dict) -> Enum:
+    """Use in enum _missing_ methods to map alternate constructions to python-legal values."""
+    try:
+        if value in mapping:
+            return mapping[value]
+        msg = f"'{value}' is not a valid {cls.__name__}"
+        raise ValueError(msg)
+    except AttributeError as _:
+        msg = f"'{value}' is not a valid {cls.__name__}"
+        raise ValueError(msg) from None
+
+
 class ApplicationDocType(str, Enum):
     """Provide values for application document type."""
 
+    AT = "at"
+    EXCLUSIVITY_LETTER = "exclusivity_letter"
+    FDA_PRESS_RELEASE = "fda_press_release"
+    FEDERAL_REGISTER_NOTICE = "federal_register_notice"
+    HEALTHCARE_PROFESSIONAL_SHEET = "healthcare_professional_sheet"
     LABEL = "label"
     LETTER = "letter"
+    MEDICATION_GUIDE = ("medication_guide",)
+    OTHER = "other"
+    OTHER_IMPORTANT_INFORMATION_FROM_FDA = "other_important_information_from_fda"
+    PATIENT_INFORMATION_SHEET = "patient_information_sheet"
+    PATIENT_PACKAGE_INSERT = "patient_package_insert"
+    PEDIATRIC_ADDENDUM = "pediatric_addendum"
+    PEDIATRIC_AMENDMENT_1 = "pediatric_amendment_1"
+    PEDIATRIC_AMENDMENT_2 = "pediatric_amendment_2"
+    PEDIATRIC_AMENDMENT_3 = "pediatric_amendment_3"
+    PEDIATRIC_AMENDMENT_4 = "pediatric_amendment_4"
+    PEDIATRIC_AMENDMENT_5 = "pediatric_amendment_5"
+    PEDIATRIC_AMENDMENT_6 = "pediatric_amendment_6"
+    PEDIATRIC_AMENDMENT_7 = "pediatric_amendment_7"
+    PEDIATRIC_CDTL_REVIEW = "pediatric_cdtl_review"
+    PEDIATRIC_CLINICAL_PHARMACOLOGY_ADDENDUM = (
+        "pediatric_clinical_pharmacology_addendum"
+    )
+    PEDIATRIC_CLINICAL_PHARMACOLOGY_REVIEW = "pediatric_clinical_pharmacology_review"
+    PEDIATRIC_DD_SUMMARY_REVIEW = "pediatric_dd_summary_review"
+    PEDIATRIC_MEDICAL_REVIEW = "pediatric_medical_review"
+    PEDIATRIC_MEMO = "pediatric_memo"
+    PEDIATRIC_OTHER = "pediatric_other"
+    PEDIATRIC_REISSUE = "pediatric_reissue"
+    PEDIATRIC_REISSUE_AMENDMENT_1 = "pediatric_reissue_amendment_1"
+    PEDIATRIC_REISSUE_AMENDMENT_2 = "pediatric_reissue_amendment_2"
+    PEDIATRIC_REISSUE_AMENDMENT_3 = "pediatric_reissue_amendment_3"
+    PEDIATRIC_REISSUE_AMENDMENT_4 = "pediatric_reissue_amendment_4"
+    PEDIATRIC_REISSUE_AMENDMENT_5 = "pediatric_reissue_amendment_5"
+    PEDIATRIC_REISSUE_AMENDMENT_6 = "pediatric_reissue_amendment_6"
+    PEDIATRIC_STATISTICAL_REVIEW = "pediatric_statistical_review"
+    PEDIATRIC_WRITTEN_REQUEST = "pediatric_written_request"
+    REMS = "rems"
     REVIEW = "review"
+    SUMMARY_REVIEW = "summary_review"
+    WITHDRAWAL_NOTICE = "withdrawal_notice"
```

Comment:
is this (and other Enum types) comprehensive? Is there reliable documentation that we can use to know what to expect for these or will we just sometimes have surprise values. (That might be the answer, I am having trouble finding documented values to expect)

Reply:
There's not reliable documentation that i'm aware of. This was constructed by manually sampling about half of all known records. We could probably do a run on the other half at some point.